packages/data-context/src/actions/TestActions.ts

```diff
@@ -0,0 +1,129 @@
+import path from 'path'
+import globby from 'globby'
+import assert from 'assert'
+import fs from 'fs-extra'
+import disparity from 'disparity'
+import tempDir from 'temp-dir'
+
+// Also defined in lib/fixtures, but we do not want to
+// import from system-tests into code we ship in production.
+const cyTmpDir = path.join(tempDir, 'cy-projects')
+
+import type { DataContext } from '..'
+import dedent from 'dedent'
+
+const systemTestsDir = path.join(__dirname, '..', '..', '..', '..', 'system-tests', 'projects')
+
+interface SnapshotScaffoldTestResult {
+  status: 'ok' | 'fail'
+  message: string
+}
+
+interface FileToDiff {
+  actual: string
+  expected: string
+}
+
+export class TestActions {
+  constructor (private ctx: DataContext) {}
+
+  async #snapshotScaffoldedFiles (expectedScaffoldDir: string, filesToDiff: FileToDiff[]): Promise<SnapshotScaffoldTestResult> {
```

Comment:
It's a new shortcut for doing `private methodName() {`... I had to look it up too. Honestly I prefer `private` just because it's more visually distinct, but absent of a coding styleguide to back me up, I was gonna keep it to myself :smile:

Reply:
I think JS lends itself poorly to OOP / classes, and the less of it we do the better our code is going to come out.

Reply:
Sure, we can just use `private`. My reasoning here was it makes more sense to use a native JS feature, which has runtime privacy guarantees, instead of a compile time only, proprietary TS feature where possible, but I don't think it makes too big a difference either way.

I don't agree private is more visually distinct though - `#` is so ugly it's impossible to not see it 😆

Reply:
I ended up moving everything into a task: https://github.com/cypress-io/cypress/pull/20818/files#diff-205e537e85adf50907257ece5985a0576754cefa95b0f691cc68bb8c11b7c3d0 which is just a function, no class, so this problem is gone.

I do think it's better to use native JS runtime features where we can over TS compile time features, though, but we can discuss that at some point.